### PR TITLE
[High level] Refactor create user

### PR DESCRIPTION
**Pull request by eleanorreem**

### CODE REVIEW

File: src/user/user.service.ts [hunk 1]
```diff
@@ -30,19 +38,20 @@ export class UserService {
...
-      email,
-      partnerAccessCode,
-      contactPermission,
-      signUpLanguage,
-      partnerId,
-      password,
-    } = createUserDto;
+    const { email, partnerAccessCode, partnerId, password } = createUserDto;
+
+    const signUpType =
```
Comment (swetha-charles) [inline src/user/user.service.ts:48]: Optional: This is a style note but I'm wondering if an `if` statement here checking on partnerId first to separate between public and partnered users would help readability. 
`if(partnerId){
 // two flavours
 } else {
// public
}
`

File: src/user/user.service.ts [hunk 2]
```diff
@@ -88,62 +98,48 @@ export class UserService {
...
-        : 'public';
+      if (signUpType === SIGNUP_TYPE.PARTNER_USER_WITHOUT_CODE) {
+        formattedUserObject = await this.createPartnerUserWithoutCode(
+          createUserDto,
+          firebaseUser.uid,
+        );
+      } else if (signUpType === SIGNUP_TYPE.PARTNER_USER_WITH_CODE) {
+        formattedUserObject = await this.createPartnerUserWithCode(createUserDto, firebaseUser.uid);
+      } else {
+        formattedUserObject = await this.createPublicUser(createUserDto, firebaseUser.uid);
```
Comment (swetha-charles) [inline src/user/user.service.ts:112]: Optional: Could check for the `SIGNUP_TYPE.PUBLIC` and then throw error in the else.

File: src/user/user.service.ts [hunk 3]
```diff
@@ -155,6 +151,102 @@ export class UserService {
...
+  ) {
+    try {
+      const createUserObject = this.userRepository.create({
+        name,
+        email,
+        firebaseUid,
+        contactPermission,
+        signUpLanguage,
+      });
+      const createUserResponse = await this.userRepository.save(createUserObject);
```
Comment (swetha-charles) [inline src/user/user.service.ts:166]: This is such a nice clean method

**Review by swetha-charles — APPROVED** [review]

File: src/user/user.service.ts (hunk 1 above)
Comment (eleanorreem) [reply src/user/user.service.ts:48]: Yeah theres a mutability vs readability conundrum!

Comment (eleanorreem) [reply src/user/user.service.ts:48]: I've switched them round so perhaps it might read easier

File: src/user/user.service.ts (hunk 2 above)
Comment (eleanorreem) [reply src/user/user.service.ts:112]: I dn't think its possible to be not one of those three types 🤔

**Pull request merged**